plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadataColumn.java

```diff
@@ -27,6 +27,7 @@
 
 public enum IcebergMetadataColumn
 {
+    PARTITION(MetadataColumns.PARTITION_COLUMN_ID, "$partition", VARCHAR, PRIMITIVE), // Avoid row type considering partition evolutions
```

Comment:
What's the problem when it's `row type`?

Reply:
The field in row type is static. It can't handle older partition spec nicely.

Reply:
(really love this new hidden column!)

a follow up question: 
would `row(c1 varchar, c2 varchar) type` work with evolutions? 
when there are other partition columns besides bucketing column, `"$partition" like '%user_bucket=1%'` won't push down.